include/pybind11/pytypes.h

```diff
@@ -2195,6 +2195,7 @@ class list : public object {
             throw error_already_set();
         }
     }
+    void clear() { PyList_SetSlice(m_ptr, 0, PyList_Size(m_ptr), nullptr); }
```

Comment:
The return value of `PyList_SetSlice()` needs to be checked, and you need to add `throw error_already_set();` if it is not `0`:

https://docs.python.org/3/c-api/list.html#c.PyList_SetSlice

(I think it's fine/best to omit a similar check for `PyList_Size()`: from the context we're sure that `m_ptr` is a Python `list` object.)